### PR TITLE
read bake definition files from remote context support

### PR DESCRIPTION
**Pull request by crazy-max**
Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: src/context.ts [hunk 1]
```diff
@@ -55,6 +57,9 @@ export async function getArgs(inputs: Inputs, buildxVersion: string): Promise<Ar
 
 async function getBakeArgs(inputs: Inputs, buildxVersion: string): Promise<Array<string>> {
   const args: Array<string> = ['bake'];
+  if (inputs.remote) {
+    args.push(inputs.remote);
```
Comment (jedevc) [inline src/context.ts:61]: Do we need special logic for this if no other opts are specified?
i.e. what happens if we have `{inputs.remote} {inputs.targets...}`?

File: README.md [hunk 2]
```diff
@@ -78,16 +78,17 @@ Following inputs can be used as `step.with` keys
...
+| `no-cache` | Bool     | Do not use cache when building the image (default `false`)                                                                                           |
+| `pull`     | Bool     | Always attempt to pull a newer version of the image (default `false`)                                                                                |
+| `load`     | Bool     | Load is a shorthand for `--set=*.output=type=docker` (default `false`)                                                                               |
+| `push`     | Bool     | Push is a shorthand for `--set=*.output=type=registry` (default `false`)                                                                             |
+| `set`      | List     | List of [targets values to override](https://github.com/docker/buildx/blob/master/docs/reference/buildx_bake.md#set) (eg: `targetpattern.key=value`) |
+| `remote`   | String   | Read bake definition files from [remote context](https://github.com/docker/buildx/blob/master/docs/guides/bake/file-definition.md#remote-definition) |
```
Comment (jedevc) [inline README.md:91]: It seems like multiple remotes are allowed? If so, should this be a `List` instead?

Comment (crazy-max) [reply README.md:91]: Only the first arg is checked as remote atm: https://github.com/docker/buildx/blob/3cf549a7f7ee015cabe99d4a08e4b9aabfc83ccc/commands/bake.go#L43

Comment (crazy-max) [reply README.md:91]: Oh actually you're right we can define remote bake files and remote context with a 2nd arg: https://github.com/docker/buildx/blob/3cf549a7f7ee015cabe99d4a08e4b9aabfc83ccc/commands/bake.go#L46-L50

**Comment by crazy-max** [comment]
> We call this "remote" internally but not sure if this is the best name for the users. Even something like "source", although overused, might be better.

Yes was not sure myself about its name. Maybe I can just use `file` and detect if the input is a remote one and place it as an arg in the cmd line?

**Comment by crazy-max** [comment]
renamed `source`. In a follow-up we could also add a `source-context` input to access the main context when a bake definition has been imported remotely with `source` input.

**Review by jedevc — APPROVED** [review]

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**